tst/EnergyPlus/unit/ZoneEquipmentManager.unit.cc

```diff
@@ -660,21 +671,196 @@ TEST_F(EnergyPlusFixture, ZoneEquipmentManager_CalcZoneMassBalanceTest3)
+        Node(ZoneEquipConfig(ZoneNum).InletNode(NodeNum)).MassFlowRate = 1.0;
+    }
+    ZoneEquipConfig(ZoneNum).InletNodeAirLoopNum(1) = 2; // Intentionally not in 1,2,3 order
+    ZoneEquipConfig(ZoneNum).InletNodeAirLoopNum(2) = 3;
+    ZoneEquipConfig(ZoneNum).InletNodeAirLoopNum(3) = 1;
+    ZoneEquipConfig(ZoneNum).ReturnNodeAirLoopNum(1) = 3; // Intentionally in a different order
+    ZoneEquipConfig(ZoneNum).ReturnNodeAirLoopNum(2) = 2;
+    ZoneEquipConfig(ZoneNum).ReturnNodeAirLoopNum(3) = 1;
+    int inletNode1 = ZoneEquipConfig(ZoneNum).InletNode(1);
+    int inletNode2 = ZoneEquipConfig(ZoneNum).InletNode(2);
+    int inletNode3 = ZoneEquipConfig(ZoneNum).InletNode(3);
+    ZoneEquipConfig(ZoneNum).ReturnNodeInletNum(1) = 2; // Intentionally in a different order
+    ZoneEquipConfig(ZoneNum).ReturnNodeInletNum(2) = 1;
+    ZoneEquipConfig(ZoneNum).ReturnNodeInletNum(3) = 3;
+    int returnNode1 = ZoneEquipConfig(ZoneNum).ReturnNode(1);
+    int returnNode2 = ZoneEquipConfig(ZoneNum).ReturnNode(2);
+    int returnNode3 = ZoneEquipConfig(ZoneNum).ReturnNode(3);
+
+    // Avoid zero values in volume flow balance check
+    DataEnvironment::StdRhoAir = 1.2;
+    DataEnvironment::OutBaroPress = 100000.0;
+    Node(ZoneEquipConfig(ZoneNum).ZoneNode).Temp = 20.0;
+    Node(ZoneEquipConfig(ZoneNum).ZoneNode).HumRat = 0.004;
+
+    DataHVACGlobals::NumPrimaryAirSys = 3;
+    DataAirSystems::PrimaryAirSystem.allocate(3);
+    DataAirLoop::AirLoopFlow.allocate(3);
+
+    // Add an outdoor air system to airloop 2
+    DataAirSystems::PrimaryAirSystem(1).OASysExists = false;
+    DataAirLoop::AirLoopFlow(1).DesReturnFrac = 1.0;
+    DataAirSystems::PrimaryAirSystem(2).OASysExists = true;
+    DataAirLoop::AirLoopFlow(2).DesReturnFrac = 0.9;
+    DataAirLoop::AirLoopFlow(2).MaxOutAir = 0.1;
+    DataAirLoop::AirLoopFlow(2).OAFlow = 0.1;
+    DataAirSystems::PrimaryAirSystem(3).OASysExists = false;
+    DataAirLoop::AirLoopFlow(3).DesReturnFrac = 1.0;
+    DataGlobals::DoingSizing = false;
+    DataGlobals::isPulseZoneSizing = false;
+
+    // Case 1 - send zero, expect zero back
+    Node(inletNode1).MassFlowRate = 0.0;
+    Node(inletNode2).MassFlowRate = 0.0;
+    Node(inletNode3).MassFlowRate = 0.0;
+    Node(returnNode1).MassFlowRate = 0.12; // Set to random values to make sure they get reset properly
+    Node(returnNode2).MassFlowRate = 0.32;
+    Node(returnNode3).MassFlowRate = 0.45;
+
+    Real64 StdTotalReturnMassFlow = 0.0;
+    Real64 FinalTotalReturnMassFlow = 0.0;
+
+    CalcZoneReturnFlows(ZoneNum, StdTotalReturnMassFlow, FinalTotalReturnMassFlow);
+    EXPECT_EQ(FinalTotalReturnMassFlow, 0.0);
+    EXPECT_EQ(Node(returnNode1).MassFlowRate, 0.0);
+    EXPECT_EQ(Node(returnNode2).MassFlowRate, 0.0);
+    EXPECT_EQ(Node(returnNode3).MassFlowRate, 0.0);
+
+    // Case 2 - send sum of inlet flow back, except system 2 back at 0.9 
+    StdTotalReturnMassFlow = 0.0;
+    FinalTotalReturnMassFlow = 0.0;
```

Comment:
@mjwitte do this only if your making changes. These 2 lines not needed. Same with 2 at line 844.

Reply:
Removed redundant lines in several places in the new test and in an older test (which is where this pattern got copied from . . . )